tsl/test/sql/compression_update_delete.sql

```diff
@@ -731,3 +731,145 @@ SELECT COUNT(*) FROM :COMPRESS_CHUNK_1;
+DELETE FROM sample_table WHERE c4 = 5;
+-- report 0 rows
+SELECT count(*) FROM sample_table WHERE c4 = 5;
+-- report 0 rows in uncompressed chunk
+SELECT COUNT(*) FROM ONLY :CHUNK_1;
+-- report 0 rows in compressed chunk
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 where c4 = 5;
+ROLLBACK;
+
+-- check that you uncompress and delete only for less than SEGMENTBY value
+BEGIN;
+-- report 50 rows
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 where c4 < 5;
+-- report 50k rows
+SELECT COUNT(*) FROM sample_table WHERE c4 < 5;
+-- delete 50k rows
+DELETE FROM sample_table WHERE c4 < 5;
+-- report 0 rows
+SELECT count(*) FROM sample_table WHERE c4 < 5;
+-- report 0 rows in uncompressed chunk
+SELECT COUNT(*) FROM ONLY :CHUNK_1;
+-- report 0 rows in compressed chunk
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 where c4 < 5;
+ROLLBACK;
+
+-- check that you uncompress and delete only for greater and equal than SEGMENTBY value
+BEGIN;
+-- report 50 rows
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 where c4 >= 5;
+-- report 50k rows
+SELECT COUNT(*) FROM sample_table WHERE c4 >= 5;
+-- delete 50k rows
+DELETE FROM sample_table WHERE c4 >= 5;
+-- report 0 rows
+SELECT count(*) FROM sample_table WHERE c4 >= 5;
+-- report 0 rows in uncompressed chunk
+SELECT COUNT(*) FROM ONLY :CHUNK_1;
+-- report 0 rows in compressed chunk
+SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 where c4 >= 5;
+ROLLBACK;
+
+-- check that you uncompress and delete only for exact ORDERBY value
+-- this will uncompress segments which have min <= value and max >= value
+BEGIN;
+-- report 10k rows
+SELECT COUNT(*) FROM sample_table WHERE c2 = 3;
+-- delete 10k rows
+DELETE FROM sample_table WHERE c2 = 3;
+-- report 0 rows
+SELECT count(*) FROM sample_table WHERE c2 = 3;
+-- report 90k rows in uncompressed chunk
+SELECT COUNT(*) FROM ONLY :CHUNK_1;
+ROLLBACK;
+
+-- check that you uncompress and delete only for less then ORDERBY value
+-- this will uncompress segments which have min < value
+BEGIN;
+-- report 20k rows
+SELECT COUNT(*) FROM sample_table WHERE c1 < 2;
+-- delete 20k rows
+DELETE FROM sample_table WHERE c1 < 2;
+-- report 0 rows
+SELECT count(*) FROM sample_table WHERE c1 < 2;
+-- report 0 rows in uncompressed chunk
+SELECT COUNT(*) FROM ONLY :CHUNK_1;
+ROLLBACK;
+
+-- check that you uncompress and delete only for greater or equal then ORDERBY value
+-- this will uncompress segments which have max >= value
+BEGIN;
+-- report 30k rows
+SELECT COUNT(*) FROM sample_table WHERE c1 >= 7;
+-- delete 30k rows
+DELETE FROM sample_table WHERE c1 >= 7;
+-- report 0 rows
+SELECT count(*) FROM sample_table WHERE c1 >= 7;
+-- report 0 rows in uncompressed chunk
+SELECT COUNT(*) FROM ONLY :CHUNK_1;
+ROLLBACK;
+
+-- check that you uncompress and delete only tuples which satisfy SEGMENTBY
+-- and ORDERBY qualifiers, segments only contain one distinct value for
+-- these qualifiers, everything should be deleted that was decompressed
+BEGIN;
+-- report 1k rows
+SELECT COUNT(*) FROM sample_table WHERE c4 = 5 and c1 = 5;
```

Comment:
Add this query before and after DELETE.
SELECT COUNT(*) FROM :COMPRESS_CHUNK_1 where c4 = 5 and _ts_meta_min_1 <= 5 and _ts_meta_max_1 >= 5;
After DELETE above query should report 0 rows.